test/hc-nested-list_test.html

```diff
@@ -42,6 +42,29 @@
         });
 
         // Public Functions
+        describe('collapseTree(parts)', () => {
+          let spy;
+
+          beforeEach(() => {
+            spy = sinon.spy(element, 'collapseTree');
+            element.parts = parts;
+          });
+
+          it('should set {open: false} on all open nodes', () => {
```

Comment:
```
it('should collapse all nodes in the tree', () => {
  Object.keys(element.parts).forEach(key =>
    element.parts[key][0].open = true);
  element.collapseTree();
  const hasOpenNodes = Object.keys(element.parts).
    some(key => element.parts[key].some(item => item.open));
  expect(hasOpenNodes).to.equal(false);
});
it('should only collapse the first node in the list of nodes', () => {
  element.parts.root[0].open = true;
  element.parts.root[1].open = true;
  element.collapseTree();
  expect(element.parts.root[0].open).to.equal(false);
  expect(element.parts.root[1].open).to.equal(true);
});
```

The first test actually ensures that all keys are checked, and not just the first item on that key, the second test ensures that it's optimized to jump out once it finds an open item, to ensure it's efficient as possible, there's another test in this file to ensure that when an item is opened that all siblings are closed.